---
title: C6312 |Microsoft 文档
ms.custom: ''
ms.date: 11/04/2016
ms.technology:
- vs-ide-code-analysis
ms.topic: conceptual
f1_keywords:
- C6312
helpviewer_keywords:
- C6312
ms.assetid: 1fc8b9a1-e6ba-4799-84c3-31f289576cca
author: mikeblome
ms.author: mblome
manager: douge
ms.workload:
- multiple
ms.openlocfilehash: 766a3c670a41bb4f915c94c3f5d0df09fde11afe
ms.sourcegitcommit: 6a9d5bd75e50947659fd6c837111a6a547884e2a
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 04/16/2018
---
# <a name="c6312"></a>C6312
警告 C6312： 可能的无限循环： 使用 try 的异常筛选器表达式中常量 EXCEPTION_CONTINUE_EXECUTION-除外  
  
 此警告意味着常量使用`EXCEPTION_CONTINUE_EXECUTION`（或另一个常数，用于计算结果为-1） 的结构化的异常处理程序的筛选器表达式中。 使用的常量值`EXCEPTION_CONTINUE_EXECUTION`可能会导致无限循环。 例如，如果硬件已引发异常，导致异常的说明将重新启动。 如果仍错误导致异常的地址，另一个异常将发生，并且处理方式相同。 这将导致无限循环。  
  
 显式调用`RaiseException`将不直接导致无限循环，但它将继续在受保护的块中的代码的执行。 这可能是意料之外，并且如果可能会导致无限循环`RaiseException`用于避免取消引用无效的指针。  
  
 通常情况下，`EXCEPTION_CONTINUE_EXECUTION`应仅由在筛选器表达式中，有机会修复导致异常的指针，或者基础内存时调用的函数。  
  
## <a name="example"></a>示例  
 下面的代码将生成此警告：  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *ptr)  
{  
  __try   
  {  
    // exception occurs if the caller passes null ptr  
    // code...     
   *ptr = '\0';  
  }   
  __except (EXCEPTION_CONTINUE_EXECUTION)  
    // When EXCEPTION_CONTINUE_EXECUTION is used, the handler  
    //  block of the structured exception handler is not executed.   
  {  
    puts("This block is never executed");  
  }  
}  
```  
  
 若要更正此警告，请使用下面的代码：  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *ptr)  
{  
  __try   
  {  
    // exception occurs if the caller passes null ptr  
    // code...  
    *ptr = '\0';  
  }   
  __except (GetExceptionCode()== EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)   
  {  
    puts("Error Occurred");  
  }  
}  
```